operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -454,6 +448,70 @@ func (r *RedpandaReconciler) reconcileClusterConfig(ctx context.Context, rp *v1a
 
 		// NB: Redpanda becoming ready and/or observing it's generation will
 		// trigger a re-queue for us.
+		return true
+	}
+
+	return false
+}
+
+func (r *RedpandaReconciler) reconcileLicense(ctx context.Context, rp *v1alpha2.Redpanda) error {
+	if r.ratelimitCondition(ctx, rp, v1alpha2.ClusterLicenseValid) {
+		return nil
+	}
+
+	client, err := r.ClientFactory.RedpandaAdminClient(ctx, rp)
+	if err != nil {
+		return err
+	}
+
+	features, err := client.GetEnterpriseFeatures(ctx)
+	if err != nil {
+		return err
+	}
+
+	var message string
+	var reason string
+	status := metav1.ConditionUnknown
+
+	switch features.LicenseStatus {
+	case rpadmin.LicenseStatusExpired:
+		status = metav1.ConditionFalse
+		reason = "LicenseExpired"
+		message = "Expired"
+	case rpadmin.LicenseStatusNotPresent:
+		status = metav1.ConditionFalse
+		reason = "LicenseNotPresent"
+		message = "Not Present"
+	case rpadmin.LicenseStatusValid:
+		status = metav1.ConditionTrue
+		reason = "LicenseValid"
+		message = "Valid"
```

Comment:
Just for future proofing and better debuggability:
```go
default:
    reason = "LicenseStatusUnknown"
    message = fmt.Sprintf("unknown license status: %q", features.LicenseStatus)
```

I've been bitten by missing defaults too many times.